### DISCUSSION TITLE
About lib\vscode\extensions\node_modules\.bin

### DISCUSSION THREAD

**Discussion by ericzhucode** [coder/code-server#5400] [General] [answered]
### Is there an existing issue for this?

- [ ] I have searched the existing issues

### OS/Web Information

- Web Browser:
- Local OS:
- Remote OS:
- Remote Architecture:
- `code-server --version`:

### Steps to Reproduce

A package issue

### Expected

 lib\vscode\extensions\node_modules\.bin should not be packaged as the path these soft link files referring to do not exist

### Actual
1. I'm not quite sure if your team have realized it and decided to keep it but just from my view, tsc & tsserver(soft link file) in lib\vscode\extensions\node_modules\.bin are meaningless and should not be packaged
![image](https://user-images.githubusercontent.com/85340915/180909531-1add083b-60c1-4670-968e-f59d814a5333.png)
As yarn run release:standalone will call postinstall.mjs script and it will delete all files in typescript lib & bin, the soft link files in screenshot refer to a non-existing path.

2. Sorry to ask more but could you tell me why we need to run yarn again in build-standalone-release.sh. Maybe it's useful in vscode source code but for built-in extension, they are packaged by webpack and all node_modules should have been added to each extension's dist folder. Why do we need to run the following command again to install node_modules?
yarn --production --frozen-lockfile --non-interactive

Sorry to disturb you guys in your busy time, looking forward to your reply

### Does this issue happen in VS Code or GitHub Codespaces?

- [ ] I cannot reproduce this in VS Code.
- [ ] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [ ] I am using HTTPS.

**Comment by jsjoeio** [comment]
@code-asher might be able to provide an answer to this

**Comment by code-asher** [comment, marked as answer]
1. Yeah maybe we should delete them.  We do delete them for the deb and rpm packages that we create through nfpm but we only did that because otherwise nfpm will fail.  We try not to mess with the upstream build too much otherwise maintenance burden increases but since we are already doing this for some packages maybe we can go ahead and do them for the others.
2. `build-standalone-release.sh` is run on different platforms so we run `yarn` again so the native modules are compiled for the right platform (for example spdlog, keytar, argon, and some others I think).  If there was a way to only recompile native modules that would be ideal but `yarn` does not seem to have that capability so we have to do a full install.

**Reply by code-asher** [reply]
Actually maybe the best way to do 2 would probably be some kind of cross-platform compilation on a single host?  Not sure if that is possible.

**Reply by code-asher** [reply]
Oh and for 1 what we should really do is submit a patch upstream to remove the broken symlinks.

**Reply by ericzhucode** [reply]
Thank you for your helpful explaination. Really appreciated!
